modules/veteran/app/models/veteran/service/representative.rb

```diff
@@ -27,8 +27,8 @@ class Representative < ApplicationRecord
       #
       # @return [Array(Veteran::Service::Representative)] All representatives found using the submitted search criteria
       def self.all_for_user(first_name:, last_name:, middle_initial: nil, poa_code: nil)
-        representatives = where('lower(first_name) = ? AND lower(last_name) = ?', first_name.downcase,
-                                last_name.downcase)
+        representatives = where('lower(first_name) = ? AND lower(last_name) = ?', first_name&.downcase,
```

Comment:
This query might not work as intended. I didn't know representatives could have nil first or last names.

In the case either field is nil, the sql generated looks like this:
```sql
SELECT
  veteran_representatives.*
FROM veteran_representatives
WHERE
  lower(first_name) = 'bob' AND lower(last_name) = NULL)
```
To get a match on "Bob <nil>",  the sql would need to be:
```sql
SELECT
  veteran_representatives.*
FROM veteran_representatives
WHERE
  lower(first_name) = 'bob' AND lower(last_name) IS NULL)
```

`= NULL` and `IS NULL` [mean different things](https://www.postgresql.org/docs/current/functions-comparison.html)

> Do not write expression = NULL because NULL is not “equal to” NULL. (The null value represents an unknown value, and it is not known whether two unknown values are equal.)

Reply:
@jvcAdHoc yep, thanks, I brought up the same issue int he shared channel in the Oddball slack org, next time i will in the lighthouse slack org. I have updated both methods to return false if either name is nil.

Reply:
Does `nil` or `[]` better match the intent of the method return? Not sure if `false` is handled well everywhere the methods are called.

Reply:
I was originally thinking adding this between 70 and 71 here https://github.com/department-of-veterans-affairs/vets-api/blob/e3b0f18b65dbfefc99e2bdbdc47c57d68592f483/modules/claims_api/app/controllers/concerns/claims_api/poa_verification.rb#L70 - no reason to kick off the call to these method if we don't have what we need. This essentially does the same thing, but my suggestion for `false` was to treat it like :74 https://github.com/department-of-veterans-affairs/vets-api/blob/e3b0f18b65dbfefc99e2bdbdc47c57d68592f483/modules/claims_api/app/controllers/concerns/claims_api/poa_verification.rb#L74